DateTimeCalendar/Date.java

```diff
@@ -0,0 +1,322 @@
+public class Date{
+    private static final int DAYS_IN_REG_YEAR = 365;
+    private static final int DAYS_IN_LEAP_YEAR = 366;
+    private static final int LEAP_YEAR_START = 1900;
+
+    public static long convToDays(int year, int month, int day){
+        int y = LEAP_YEAR_START;
+        Month m = Date.Month.values()[month - 1];
+        long payload = (long)day;
+
+        //Process Year Value
+        if (year >= LEAP_YEAR_START){
+            while (y < year){
+               if (isLeapYear(y)){
+                    payload += (long)DAYS_IN_LEAP_YEAR;
+                } else {
+                    payload += (long)DAYS_IN_REG_YEAR;
+                } 
+                y++;
+            }
+        } else {
+            throw new IllegalArgumentException("Invalid Year entered");
+        }
+
+        // //Process Month Value
+        for (int i = 0; i < month - 1; i++){
+            if(isLeapYear(year) && Date.Month.values()[i] == Month.FEB){
+                payload += Date.Month.values()[i].days + 1;
+            } else {
+                payload += Date.Month.values()[i].days;
+            }
+        }
+
+        return payload;
+    }
+
+    public static boolean isLeapYear(int year){
+        return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+    }
```

Comment:
This method should also be provided at the class level using the current year defined in the instance. So you can do:
```java
Date dt = new Date(...);
dt.isLeapYear();
```

Reply:
I've added this in with the other methods. I wasn't sure if it belong there or with the getters and setters, since I see logic in having it in both areas.